integration/nodejs_stack_integration_test.go

```diff
@@ -0,0 +1,352 @@
+		})
+
+		it("sucessfully builds an app", func() {
+
+			image, _, err = pack.Build.
+				WithExtensions(
+					settings.Extensions.UbiNodejsExtension.Online,
+				).
+				WithBuildpacks(
+					settings.Buildpacks.NodeEngine.Online,
+					settings.Buildpacks.NPMInstall.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithBuilder(builder).
+				WithNetwork("host").
+				WithEnv(map[string]string{"BP_NODE_RUN_EXTENSION": runNodejs18ImageID}).
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithPublish("8080").
+				WithCommand("node server.js").
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(container).Should(Serve("Hello World!"))
+		})
+	})
+
+	context("When using Node.js 20 as run image", func() {
+		var (
+			err                error
+			builder            string
+			buildImageID       string
+			runImageID         string
+			runNodejs20ImageID string
+			nodeMajorVersion   int
+		)
+
+		it.After(func() {
+			Expect(docker.Image.Remove.Execute(builder)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(buildImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runNodejs20ImageID)).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it.Before(func() {
+			nodeMajorVersion = 20
+
+			buildImageID, runImageID, builderConfigFilepath, builder, err = generateBuilder(root)
+			Expect(err).NotTo(HaveOccurred())
+
+			source, err = occam.Source(filepath.Join("testdata", "nodejs_simple_app"))
+			Expect(err).NotTo(HaveOccurred())
+
+			//Creating and pushing the run image to local registry
+			runNodejs20Archive := filepath.Join(root, fmt.Sprintf("../build-nodejs-%d", nodeMajorVersion), "run.oci")
+			runNodejs20ImageID = fmt.Sprintf("run-nodejs-%d-%s", nodeMajorVersion, uuid.NewString())
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(archiveToDaemon(runNodejs20Archive, runNodejs20ImageID)).To(Succeed())
+		})
+
+		it("sucessfully builds an app", func() {
+
+			image, _, err = pack.Build.
+				WithExtensions(
+					settings.Extensions.UbiNodejsExtension.Online,
+				).
+				WithBuildpacks(
+					settings.Buildpacks.NodeEngine.Online,
+					settings.Buildpacks.NPMInstall.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithBuilder(builder).
+				WithNetwork("host").
+				WithEnv(map[string]string{"BP_NODE_RUN_EXTENSION": runNodejs20ImageID}).
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithPublish("8080").
+				WithCommand("node server.js").
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(container).Should(Serve("Hello World!"))
+		})
+	})
+
+}
+
+func archiveToDaemon(path, id string) error {
```

Comment:
I'm wondering if there is a better name for this? I guess its coping an oci-archive into the damon storage or local registry?

Reply:
Is in daemon storage, they also use this name in jammy, https://github.com/paketo-buildpacks/jammy-base-stack/blob/f96f2f10ee3a438657442763bae24ee8ba9bbaf7/buildpack_integration_test.go#L143 So I kept it mostly for consistency.

Reply:
@pacostas, makes sense if that name is used in the other stacks 👍